<launch>
	<!-- base params -->
	<arg name="mav_addr" default="udpin:0.0.0.0:14551" />
	<arg name="world_frame" default="world" />
	<!-- <arg name="rov_frame" default="orb_slam3_camera" /> -->
	<arg name="rov_frame" default="bluerov2/base_link" />
	<arg name="is_sim" default="True" />

	<!-- optional set home -->
	<arg name="set_home" default="true" />
	<arg name="home_lat" default="24.52433290904017" />
	<arg name="home_long" default="54.43519869059818" />
	<arg name="home_alt" default="0" />

	<arg name="log_output" default="screen" />
	<arg name="respawn_viso" default="false" />

	<node pkg="controls" type="send_viso_tf2.py" name="send_viso_tf2" required="$(eval not respawn_viso)"
	    clear_params="true" output="$(arg log_output)" respawn="$(arg respawn_viso)">
		<param name="mav_addr" value="$(arg mav_addr)" />
		<param name="world_frame" value="$(arg world_frame)" />
		<param name="rov_frame" value="$(arg rov_frame)" />
		<param name="set_home" value="$(arg set_home)" />
		<param name="home_lat" value="$(arg home_lat)" />
		<param name="home_long" value="$(arg home_long)" />
		<param name="home_alt" value="$(arg home_alt)" />
		<param name="is_sim" value="$(arg is_sim)" />
	</node>

	<!-- Static broadcasters for IMU <-> camera transforms courtesy of:
	 	http://wiki.ros.org/tf2/Tutorials/Writing%20a%20tf2%20static%20broadcaster%20%28C%2B%2B%29
		https://www.andre-gaschler.com/rotationconverter/ -->
	<!-- <node pkg="tf2_ros" type="static_transform_publisher" name="cam_imu_broadcaster" args="0.2 0 0 0.5 0.5 0.5 0.5 base_link pcl_link" /> -->
</launch>